### PR TITLE
#399 fix headings and css improvements

### LINKED ISSUE #399: Fix headings on pages
h2's are used on most pages. These should be h1's in most cases: https://www.a11yproject.com/posts/how-to-accessible-heading-structure/#best-practices-summarized

This will allow for better structured html which will benefit accessibility tools like screen readers

### PR DESCRIPTION
**Pull request by bmpickford**
### Pull Request Checklist

* [x] Pull request is based on the `main` branch
* [x] Pull request includes a [sign off]

Fixes #399 

Signed-off-by: Benjamin Pickford b.mpickford@outlook.com

### Details
Apologies, I got a bit carried away. Most of the changes are changing a colour to use a css var or changing h2 to h1

 * Turned most of the h2's into h1's, and the h3's into h2's so the headings are structured more appropriately
     * Tried to follow the best practices outlined here: https://www.a11yproject.com/posts/how-to-accessible-heading-structure/#best-practices-summarized
     * This should slightly help screen readers
     * Will also help SEO but probably don't care about that here
 * Added some basic css vars for reuse based on https://www.figma.com/file/omAtzW5Kfg7NN97ECuQ3GC/Govhack-Hackerspace-Reskin?type=design&node-id=695-404&t=7MPph7EA11xm7Kid-0
 * Consolidated heading styles
 * Tried to make the pages align more and removed duplicate/unneeded css where I could

### Evidence of testing
Obviously doesn't cover everything but covers the pages I did more bespoke css changes on

https://github.com/govhackaustralia/hackerspace3/assets/12684286/9dad7c12-8114-457b-93fb-86be1b00a0bd

### CODE REVIEW

File: app/controllers/data_sets_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class DataSetsController < ApplicationController
   def index
     @data_sets = @competition.data_sets.order(:name).preload(:region)
+    @regions = @data_sets.map(&:region).uniq
```
Comment (astley92) [inline app/controllers/data_sets_controller.rb:4]: Just wondering the purpose of this addition? I don't see this being used in the index view is all?

Also, more of a drive by comment and not related to this PR, but IMO it'd be awesome to start moving away from the excessive usage of instance variables to pass data to views. Another option is to use render locals to achieve the same thing without instance vars. 

There's a bunch of reasons I'd be happy to chat about but one of the main reasons would be that attempting to access an undefined instance var results in nil which can lead to some hard to debug exception output. Whereas passing variables, you'll get a more logical explanation of what went wrong.

**Review by astley92 — APPROVED** [review]
Nice LGTM! Only minor nitpicks and a comment that is more focused around starting a discussion that actioning anything here

File: app/controllers/data_sets_controller.rb (hunk 1 above)
Comment (bmpickford) [reply app/controllers/data_sets_controller.rb:4]: Ah yep, this was leftover from something else I was doing. Will delete

File: app/assets/stylesheets/components/top_nav_bar.scss [hunk 2]
```diff
@@ -1,5 +1,5 @@
 .top-bar {
-  box-shadow: -8px 6px 30px -4px rgba(0, 0, 0, 0.55);
+  // box-shadow: -8px 6px 30px -4px rgba(0, 0, 0, 0.55);
```
[the remark this answers is not in the document]
Comment (bmpickford) [reply app/assets/stylesheets/components/top_nav_bar.scss:2]: Nope, will delete

**Pull request merged**